### PR TITLE
Update to TileDB 2.7 and support all new features and changes

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**
This PR updates TileDB-Java to support the latest TileDB-2.7 version. 
It also adds all the new improvements & features which are:
-[New Subarray API exposure](https://github.com/TileDB-Inc/TileDB/pull/2214)
-Support for the refactored dense reader
-New methods: `tiledb_array_schema_get_version`, `tiledb_array_schema_evolution_set_timestamp_range`, `tiledb_array_schema_evolution_get_timestamp_range`, `tiledb_fragment_info_set_config`

### CODE REVIEW

File: cmake/Modules/FindTileDB_EP.cmake [hunk 1]
```diff
@@ -48,14 +48,14 @@ if (NOT TILEDB_FOUND)
     # Try to download prebuilt artifacts unless the user specifies to build from source
     if(DOWNLOAD_TILEDB_PREBUILT)
         if (WIN32) # Windows
-          SET(DOWNLOAD_URL "https://github.com/TileDB-Inc/TileDB/releases/download/2.6.3/tiledb-windows-x86_64-2.6.3-534ca7c.zip")
-          SET(DOWNLOAD_SHA1 "bdfb3841b3c85d4fb1ee64f74d2d54eb39e1a3d5")
+          SET(DOWNLOAD_URL "https://github.com/TileDB-Inc/TileDB/releases/download/2.7.0/tiledb-windows-x86_64-2.7.0-4f39c6c.zip")
```
Comment (ihnorton) [inline cmake/Modules/FindTileDB_EP.cmake:51]: Doesn't need to be updated here, but FYI 2.7.1 is available.

File: src/test/java/io/tiledb/java/api/QueryConditionTest.java [hunk 2]
```diff
@@ -55,14 +55,14 @@ public void setup() throws Exception {
...
-        new Dimension<Long>(ctx, "d2", Long.class, new Pair<Long, Long>(1l, 3l), 2l);
+        new Dimension<Long>(ctx, "d1", Long.class, new Pair<Long, Long>(1l, 9l), 2l);
+    //    Dimension<Long> d2 =
+    //        new Dimension<Long>(ctx, "d2", Long.class, new Pair<Long, Long>(1l, 3l), 2l);
 
     // Create and set getDomain
     Domain domain = new Domain(ctx);
     domain.addDimension(d1);
-    domain.addDimension(d2);
+    //    domain.addDimension(d2);
```
Comment (ihnorton) [inline src/test/java/io/tiledb/java/api/QueryConditionTest.java:65]: why removed, here and above?

**Review by ihnorton — APPROVED** [review]
Minor ?s. LGTM, thank you!

File: cmake/Modules/FindTileDB_EP.cmake (hunk 1 above)
Comment (DimitrisStaratzis) [reply cmake/Modules/FindTileDB_EP.cmake]: Updated to 2.7.1!

File: src/test/java/io/tiledb/java/api/QueryConditionTest.java (hunk 2 above)
Comment (DimitrisStaratzis) [reply src/test/java/io/tiledb/java/api/QueryConditionTest.java]: I was testing something and forgot to revert to the original. Should be ok now.

File: cmake/Modules/FindTileDB_EP.cmake (hunk 1 above)
Comment (DimitrisStaratzis) [reply cmake/Modules/FindTileDB_EP.cmake]: Also, what do you mean, it doesn't need to be updated here?

Comment (ihnorton) [reply cmake/Modules/FindTileDB_EP.cmake]: resolving

**Pull request merged**